### ISSUE TITLE
Sleep Timer Presents

### ISSUE THREAD

**Issue by Fofer** [SeydX/homebridge-bravia-tvos#103]
Launching the "Timers" app displays this, which is great, a step in the right direction!

![IMG_6613](https://user-images.githubusercontent.com/518749/99917669-9206d480-2cc6-11eb-9534-fcdc5904f834.jpg)

But then I need to switch to the Sony remote to actually select one of the options.  So I was thinking, how great would it be if these options could be exposed as preset commands, in HomeKit?  Gamechanger!

**Comment by SeydX** [comment]
Added in 4.1.2

[Example Config](https://github.com/SeydX/homebridge-bravia-tvos/blob/master/example-config.json#L69)

**Comment by Fofer** [comment]
This is fantastic!  Again, thanks so much. I am now trying to create some working "sleep timer" macros, that don't exit the screen I'm watching.

Unfortunately, at least on my my Bravia TV model, invoking the Sleep timer isn't possible from the Action/Top Menu.

I don't see a way to launch the "Timers" application via a macro command, I can only launch it by adding it under "Applications."

I tested some other commands, but the "Sleep" command puts the TV to sleep immediately and "SleepTimer" command just turns off the Sleep Timer off.

If I can somehow launch the Timers app within the macro, I can handle the rest via "Up" and "Confirm" commands.
I do see some apps listed there, but not the Timers app. And I see "PartnerApp1" all the way through "PartnerApp16."  Can these be customized, or is there a way to add "Timers" to the available list of apps/commands that can be strung together in a macro?

**Comment by SeydX** [comment]
Hello @Fofer 

The custom ui offers you all the applications and commands your TV can handle. If there is no application or command to open the timer list, there is no way for me to create one.

**Comment by Fofer** [comment]
Yes, perhaps my wording wasn't clear.  The application is called "Timers."

This application appears in the Applications list. I can invoke it from there, and it works.  That's how I got to the screenshot in the first post of this report.

![CleanShot 2020-11-23 at 06 48 07](https://user-images.githubusercontent.com/518749/99976177-f756d600-2d57-11eb-96c8-dd2ed34f8ed1.png)

This application does NOT appear in the Commands list so I am not able to string it together in a Macro.

![CleanShot 2020-11-23 at 06 47 06](https://user-images.githubusercontent.com/518749/99976212-ff167a80-2d57-11eb-99de-9f7e4c0c6fdf.png)

What are these PartnerApps? Can I use one of these slots to launch the "Timers" app?

**Comment by SeydX** [comment]
@Fofer 

It seems your tv has also ircc commands to open partner. What kind of apps these are idk.

Thats nothing what i have coded. The plugin just lists everything what your tv offers

**Comment by Fofer** [comment]
Ok I understand.

So is there a way for me to create a macro that first launches the Timers app, and then issues these commands : up + up + enter?

Because right now the only way I can do is is with two separate macros.

**Comment by Fofer** [comment]
Hi, just to add more information in the meantime for anyone following along. I've managed to use the iOS shortcuts app to string together the separate commands that this plugin offers to HomeKit.

The first step of my automation sends the "Launch Timers App" command to the Sony TV.  Then it waits a second, and sends the Macro command sequence of "down, DpadCenter, Up, Up, DpadCenter, TV, Sleep Timer."   The last two commands exit the Timers app and display the Sleep status in a label at the bottom of the screen.  

It would be great if "app launches" could be included with "Macro" command sequences via this Homebridge plugin to start, with those combined macros available directly in the Home app along with the others listed inside the Sony TV appliance, but saving this combination shortcut to my iOS home screen is a suitable alternative, too.

Thanks again for your great work on this, @SeydX.

**Comment by SeydX** [comment]
Maybe I will revise the macros for the next version so that you can also open apps/channels and even set the "delay" between commands

**Comment by Fofer** [comment]
Oh wow that would be AMAZING! It would open up all sorts of new possibilities because then macros could easily be used inside apps we launch. Thanks for the consideration and your excellent continued work on this absolute game-changer of a plugin. Happy holidays.
